### ISSUE TITLE
[BUG] $ in env file

### ISSUE THREAD

**Issue by vanodevium** [docker/compose#11058]
### Description

If there is a `$` in the value, it is treated as a variable. Even if wrapped in quotas. Only escaping help. But for another languages and projects, `$` is treated as a symbol.

### Steps To Reproduce

1. .env file
```sh
PASSWORD="abc$def"
```

2. compose file
```yml
version: "3.9"

services:
  static0:
    image: "caddy:2.7.4-alpine"
    container_name: "static0"
    restart: unless-stopped
    env_file: .env
    ports:
      - "81:80"
```

3. PASSWORD in the container would be `abc`

### Compose Version

```Text
Docker Compose version v2.22.0
```

### Docker Environment

```Text
Client: Docker Engine - Community
 Version:    24.0.6
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.11.2
    Path:     /usr/libexec/docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.21.0
    Path:     /usr/libexec/docker/cli-plugins/docker-compose
```

**Comment by ndeloof** [comment]
See https://github.com/compose-spec/compose-spec/blob/master/05-services.md#env_file-format for the dot env file format.

tl;dr: use single quotes to disable variable interpolation
